reentrancy_guard_example/deposit_example/src/deposit_actor.rs

```diff
@@ -0,0 +1,264 @@
+
+        let st = State::new(rt.store()).map_err(|e| {
+            e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "Failed to create actor state.")
+        })?;
+
+        rt.create(&st)?;
+
+        Ok(())
+    }
+
+    fn balance(rt: &impl Runtime) -> Result<TokenAmount, ActorError> {
+        rt.validate_immediate_caller_accept_any()?;
+
+        Ok(rt.current_balance())
+    }
+
+    fn deposit(rt: &impl Runtime) -> Result<(), ActorError> {
+        rt.validate_immediate_caller_accept_any()?;
+
+        let value = rt.message().value_received();
+        let sender = Actor::msg_sender();
+
+        rt.transaction(|st: &mut State, rt| {
+            let mut map = map::make_map_with_root(&st.deposit_funds, rt.store()).map_err(|e| {
+                e.downcast_default(
+                    ExitCode::USR_ASSERTION_FAILED,
+                    "Failed to create map from Cid.",
+                )
+            })?;
+
+            let key: BytesKey = sender.to_bytes().into();
+
+            if let Some(val) = map.get(&key).map_err(|e| {
+                e.downcast_default(
+                    ExitCode::USR_ASSERTION_FAILED,
+                    "Failed to check senders balance.",
+                )
+            })? {
+                map.set(key, value.clone() + val).map_err(|e| {
+                    e.downcast_default(
+                        ExitCode::USR_ASSERTION_FAILED,
+                        "Failed to add deposit to map.",
+                    )
+                })?;
+            } else {
+                map.set(key, value.clone()).map_err(|e| {
+                    e.downcast_default(
+                        ExitCode::USR_ASSERTION_FAILED,
+                        "Failed to add deposit to map.",
+                    )
+                })?;
+            }
+
+            st.deposit_funds = map.flush().map_err(|e| {
+                e.downcast_default(ExitCode::USR_ASSERTION_FAILED, "Failed to flush map.")
+            })?;
+
+            Ok(())
+        })
+    }
+
+    fn check_balance(rt: &impl Runtime, address: Address) -> Result<TokenAmount, ActorError> {
+        rt.validate_immediate_caller_accept_any()?;
+
+        let address = match resolve_address(&address) {
+            Some(address) => address,
+            None => {
+                return Err(actor_error!(
+                    assertion_failed,
+                    "Err: cannot resolve address"
+                ))
+            }
+        };
+
+        let st: State = rt.state()?;
+
+        let map = map::make_map_with_root::<_, TokenAmount>(&st.deposit_funds, rt.store())
+            .map_err(|e| {
+                e.downcast_default(
+                    ExitCode::USR_ASSERTION_FAILED,
+                    "Failed to create map from Cid.",
+                )
+            })?;
+
+        let key: BytesKey = address.to_bytes().into();
+
+        match map.get(&key).unwrap() {
+            Some(val) => Ok((*val).clone()),
```

Comment:
You can use .to_owned() instead, try something like
```suggestion
            Some(val) => Ok(val.to_owned()),
```

Reply:
Changed [here](https://github.com/Ethernal-Tech/filecoin-actor-library/blob/c644a0b2285365beb966794427a1877ff545db94/reentrancy_guard_example/deposit_example/src/deposit_actor.rs#L139).